core/src/androidMain/kotlin/Peripheral.kt

```diff
@@ -193,22 +193,22 @@ public class AndroidPeripheral internal constructor(
 
     /** Creates a connect [Job] that completes when connection is established, or failure occurs. */
     private fun connectAsync() = scope.async(start = LAZY) {
-        val connection = establishConnection().also { _connection = it }
+        try {
+            val connection = establishConnection().also { _connection = it }
 
-        connection
-            .characteristicChanges
-            .onEach(observers.characteristicChanges::emit)
-            .launchIn(scope, start = UNDISPATCHED)
+            connection
+                .characteristicChanges
+                .onEach(observers.characteristicChanges::emit)
+                .launchIn(scope, start = UNDISPATCHED)
```

Comment:
Is it less about moving the `connection...launchIn` portion inside the `try` and more about moving the `establishConnection`?

I just want to understand the rationale. Line 199-202 won't throw an exception, at least not that I can see (it would be propagated to the `scope` anyways).

But I can see the `establishConnection` might throw, in which case getting the log statement could be helpful.

Reply:
> Is it less about moving the connection...launchIn portion inside the try and more about moving the establishConnection?

Yup. The launch just got caught up in moving it. I could put a separate try-catch around the `establishConnection` if you'd prefer.

Reply:
> Yup. The launch just got caught up in moving it. I could put a separate try-catch around the `establishConnection` if you'd prefer.

Naw, no need. Was just inquiring to understand.

I think it is fine how you have it now.